code/game/world.dm

```diff
@@ -47,6 +47,10 @@ GLOBAL_PROTECT(security_mode)
 	if(TEST_RUN_PARAMETER in params)
 		HandleTestRun()
 
+/world/Del()
+	call(RUST_G, "log_close_all")()
+	..()

```

Comment:
We don't use /Del() because it might incur overhead on all deletions. Use the subsystem shutdown system instead

Reply:
I'm not sure if there's a better place for this really. Will `/world/Reboot` (which is the only thing which invokes `master.Shutdown()`) always run? This needs to run as late as is possible.

Reply:
I recommend adding some new proc shutdown_native_code which is called after all byond code shutdowns have complete (save Del) this should explicitly have a headsup that no ss13 code can be relied upon to run correclty at the point of calling this